python/ray/experimental/workflow/execution.py

```diff
@@ -38,28 +59,23 @@ def run(entry_workflow: Workflow,
     if workflow_id is None:
         # Workflow ID format: {Entry workflow UUID}.{Unix time to nanoseconds}
         workflow_id = f"{entry_workflow.id}.{time.time():.9f}"
-    if isinstance(storage, str):
-        set_global_storage(create_storage(storage))
-    elif isinstance(storage, Storage):
-        set_global_storage(storage)
-    elif storage is not None:
-        raise TypeError("'storage' should be None, str, or Storage type.")
-    storage_url = get_global_storage().storage_url
+    store = _get_storage(storage)
     logger.info(f"Workflow job created. [id=\"{workflow_id}\", storage_url="
-                f"\"{storage_url}\"].")
-    try:
-        workflow_context.init_workflow_step_context(workflow_id, storage_url)
-        commit_step(entry_workflow)
```

Comment:
I like the commit step api here since it put the commit code into one explicitly. Any reason why you get rid of it?

Reply:
It is because the function seems too "shallow", it only has 3 lines of code. After addressing the bug, the number of arguments are more than the lines of code.

Reply:
I'm also fine if you decide to keep it. I think it is useful when we pass storage via context.

Reply:
Let's keep commit_step here which makes it easier to understand. Besides, there is also another place commit the step.